docs/tutorials/basics/50_control_flow.md

```diff
@@ -240,3 +240,56 @@ end
 
 Both clauses have branches with the same conditions but in a different order and they behave differently.
 The first matching condition selects which branch executes.
+
+## Iteration
+
+Crystal offers several ways to do iteration. If iterating over an _iterable_ object, like an array, a natural way is to use `each`:
+
+```crystal-play
+vals = ["one", "two", "three", "four"]
+vals.each { |num| puts num }
+```
+
+It is also possible to get the index using `each_With_index`:
+
+```crystal-play
+vals = ["one", "two", "three", "four"]
+vals.each_with_index { |num, id| puts "#{num} is #{id}" }
+```
+
+`upto` allows iterations in a range:
```

Comment:
What do you think of starting this part with an example of `(1..5).each` to introduce a range, and then this one? Just a thought.

Reply:
I was meaning to use range's `each` `(1..5).each` prior to `upto`, but somehow the reordering does flow nicely now, so 👍 👍